llm-examples/mpt/mpt-7b/scripts/fine_tune_deepspeed.py

```diff
@@ -0,0 +1,220 @@
+import click
+from functools import partial
+import json
+import logging
+import os
+import numpy as np
+from pathlib import Path
+import torch
+
+from datasets import Dataset, load_dataset
+import transformers
+from transformers import (
+    AutoConfig,
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    DataCollatorForLanguageModeling,
+    PreTrainedTokenizer,
+    Trainer,
+    TrainingArguments,
+    set_seed,
+)
+
+
+os.environ['HF_HOME'] = '/local_disk0/hf'
+os.environ['TRANSFORMERS_CACHE'] = '/local_disk0/hf'
+
+logger = logging.getLogger(__name__)
+
+ROOT_PATH = Path(__file__).parent.parent
+MODEL_PATH = 'mosaicml/mpt-7b'
+TOKENIZER_PATH = 'EleutherAI/gpt-neox-20b'
+DEFAULT_TRAINING_DATASET = "databricks/databricks-dolly-15k"
+CONFIG_PATH = "../../config/a10_config.json"
+LOCAL_OUTPUT_DIR = "/local_disk0/output"
+DEFAULT_SEED = 68
+
+
+def load_training_dataset(
+  tokenizer,
+  path_or_dataset: str = DEFAULT_TRAINING_DATASET,
+  ) -> Dataset:
+    logger.info(f"Loading dataset from {path_or_dataset}")
+    dataset = load_dataset(path_or_dataset)
+    logger.info(f"Found {dataset.num_rows} rows", )
+
+    def _reformat_data(rec):
```

Comment:
It would be helpful if there's comment on why this is needed, for example it is required by databricks-dolly-15k.